src/rgw/driver/rados/rgw_trim_bilog.cc

```diff
@@ -759,14 +904,45 @@ int BucketTrimInstanceCR::operate(const DoutPrefixProvider *dpp)
 			    << cpp_strerror(retcode) << dendl;
 	  return set_cr_error(retcode);
 	}
+
+  if (clean_info->first.layout.logs.front().layout.type == rgw::BucketLogType::Deleted) {
+    if (list_results.objs.empty()) {
+      //remove bucket instance metadata
+      yield call(new RGWRemoveBucketInstanceInfoCR(
+      store->svc()->async_processor,
+      store, clean_info->first.bucket,
+      clean_info->first, nullptr, dpp));
+      if (retcode < 0) {
+        ldpp_dout(dpp, 0) << "failed to remove instance bucket info: "
+                          << cpp_strerror(retcode) << dendl;
+        return set_cr_error(retcode);
+        }
+    } else {
+      yield {
+        // if there are objects present, add the bucket entry to omap list 'deleted_buckets'
+        auto& pool = store->svc()->zone->get_zone_params().log_pool;
+        const bufferlist in_bl;
+        std::map<std::string, bufferlist> bucket_entry{{bucket_instance, in_bl}};
+        ldpp_dout(dpp, 10) << "storing bucket in omap: " << bucket_instance << dendl;
+        call(new RGWRadosSetOmapKeysCR(store, rgw_raw_obj(pool, oid_prefix), bucket_entry));
```

Comment:
does this `bucket_instance` string contain the bucket name, or just the bucket instance id?

to complete this cleanup, i think we'd expect the admin to run a command like:
```
$ radosgw-admin bucket rm --bucket {name} --bucket-id {instance-id} --purge-objects
```
which requires the bucket name too

Reply:
it has tenant, bucket name and id. like this:

```
[
    "testx/bucket100:96c7bca1-54be-49e9-9cf2-ebbbd0153aea.4451.102",
    "testx/bucket102:96c7bca1-54be-49e9-9cf2-ebbbd0153aea.4451.104"
]
```